### PR TITLE
Support stable OTel deployment environment attribute

### PR DESCRIPTION
**Pull request by bm1549**
## Summary of changes

Map `deployment.environment.name` to the Datadog environment. Keep `deployment.environment` as a fallback, and keep `DD_ENV` as the highest-precedence source.

## Reason for change

OpenTelemetry stabilized the deployment environment semantic convention under `deployment.environment.name`.

Related PRs:

- DataDog/system-tests#7613
- DataDog/dd-trace-go#5285
- DataDog/dd-trace-dotnet#9150
- DataDog/dd-trace-rb#6261
- DataDog/dd-trace-php#4148
- DataDog/dd-trace-js#10050
- DataDog/dd-trace-rs#300
- DataDog/dd-trace-java#12324
- DataDog/dd-trace-py#19901

## Implementation details

The resource attribute mapper tracks the stable and legacy values separately, then chooses the stable value regardless of input order. It removes both source aliases after promotion and leaves unrelated attributes alone.

## Test coverage

Five focused mapping cases and 182 adjacent configuration tests passed locally. Formatting checks also passed.

## Other details

None.

### CODE REVIEW

File: tracer/src/Datadog.Trace/Configuration/MutableSettings.cs [hunk 1]
```diff
@@ -1107,8 +1107,10 @@ private static ConfigurationBuilder.ClassConfigurationResultWithKey<IDictionary<
         if (original.ConfigurationResult is { IsValid: true, Result: { } values })
         {
             // Update well-known service information resources
-            if (values.TryGetValue("deployment.environment", out var envValue))
+            if (values.TryGetValue("deployment.environment.name", out var envValue) ||
+                values.TryGetValue("deployment.environment", out envValue))
```
Comment (bouwkast) [inline tracer/src/Datadog.Trace/Configuration/MutableSettings.cs:1111]: Could we add a test for what happens with an empty string for `deployment.environment.name` ?

**Comment by zacharycmontoya** [comment]
There are going to be some small merge conflicts so let's wait to include this after the next release, when we'll re-incorporate Milan's tracestate work

File: tracer/src/Datadog.Trace/Configuration/MutableSettings.cs (hunk 1 above)
Comment (zacharycmontoya) [reply tracer/src/Datadog.Trace/Configuration/MutableSettings.cs]: I've added a unit test for the empty `deployment.environment.name`. I also updated the behavior to a reasonable behavior where we fall back to the legacy one if the new stable one is null or empty. Done in 08968bcb78ca4eaadf574e39a4e0697bb85d76f4

Comment (bouwkast) [reply tracer/src/Datadog.Trace/Configuration/MutableSettings.cs]: I think if we make this change it needs to be synchronized across libraries as we now differ

I think this is the correct approach though

Maybe https://github.com/DataDog/system-tests/pull/7613 needs to be updated to test / assert the behavior for empty values

I won't block on this though

Comment (bouwkast) [reply tracer/src/Datadog.Trace/Configuration/MutableSettings.cs]: Interestingly the Rust PR follows this new behavior

But the merged Java/Python/Node.js PRs use the behavior that this PR initially had

So I think we may want to figure out what route to go

Comment (zacharycmontoya) [reply tracer/src/Datadog.Trace/Configuration/MutableSettings.cs]: Yeah I also checked and found the same thing. Since this is only an edge case I'd like to merge this PR as-is to have the common case implemented, then we can align the other tracers separately

**Review by bouwkast — APPROVED** [review]

**Review by zacharycmontoya — APPROVED** [review]
Approving the improvement, but I also added a commit to this PR so does it count?
